connect/connect.go

```diff
@@ -18,7 +18,7 @@ type ConnectService interface {
 
 	// Acknowledge acknowledges a response sent by a device and returns
 	// the next payload if one is available.
-	Acknowledge(ctx context.Context, req mdm.Response) (payload []byte, err error)
+	Acknowledge(ctx context.Context, req mdmConnectRequest) (payload []byte, err error)
```

Comment:
`mdmConnectRequest` is a private type and can't be used in the function signature. 
It's also a endpoint specific wrapper type, not intended to be exposed. 

I'd be more willing to consider a type like 

```type Response struct {
    Raw []byte
    MDMResponse mdm.Response
}
````